x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -745,30 +749,47 @@ class AgentPolicyService {
           }),
     };
 
+    const permissions = (await storedPackagePoliciesToAgentPermissions(
+      soClient,
+      agentPolicy.package_policies
+    )) || { _fallback: DEFAULT_PERMISSIONS };
+
+    permissions._elastic_agent_checks = {
+      cluster: DEFAULT_PERMISSIONS.cluster,
+    };
+
+    // TODO fetch this from the elastic agent package
+    const monitoringOutput = fullAgentPolicy.agent?.monitoring.use_output;
+    if (
+      fullAgentPolicy.agent?.monitoring.enabled &&
+      monitoringOutput &&
+      fullAgentPolicy.outputs[monitoringOutput]?.type === 'elasticsearch'
+    ) {
+      const names: string[] = [];
+      if (fullAgentPolicy.agent.monitoring.logs) {
+        names.push('logs-*');
+      }
+      if (fullAgentPolicy.agent.monitoring.metrics) {
+        names.push('metrics-*');
+      }
```

Comment:
Please review [my comment](https://github.com/elastic/kibana/pull/97366#issuecomment-848862495) again, these aren't the patterns to use, they should be:
`logs-elastic_agent.*-{AGENT POLICY NAMESPACE}`
`metrics-elastic_agent.*-{AGENT POLICY NAMESPACE}`